Horan_Colby/JS_Fundamentals3/part3.js

```diff
@@ -0,0 +1,77 @@
+var person = {
+	name : 'Colby',
+	distance_travelled : 0,
+	say_name : function(){ console.log(person.name) },
+	say_something : function(something){
+		console.log(person.name, + 'says' + something)
+	},
+	walk : function(){
+		console.log(person.name, 'is walking')
+		person.distance_travelled += 3
+	},
+	run : function(){
+		console.log(person.name, 'is running')
+		person.distance_travelled += 10
+	},
+	crawl : function(){
+		console.log(person.name, 'is crawling')
+		person.distance_travelled += 1
+	}
+}
+
+function personConstructor(name){
+	var person = {
+		name : name,
+		distance_travelled : 0,
+		say_name : function(){ console.log(person.name) },
+		say_something : function(something){
+			console.log(person.name + ' says ' + something) },
+		walk : function(){
+			console.log(person.name + ' is walking')
+			person.distance_travelled += 3 },
+		run : function(){
+			console.log(person.name + ' is running')
+			person.distance_travelled += 10 },
+		crawl : function(){
+			console.log(person.name + ' is crawling')
+			person.distance_travelled += 1 }
+	}
+	console.log(person.name)
+	person.say_something('YAS YAS YAS YAS')
+}
+
+
+function ninjaConstructor(name){
+	var ninja = {
+		name : name,
+		cohort : 'September 19th, 2016',
+		belt : 'yellow-belt',
+		levelUp : function(){
```

Comment:
A cool way to do this is to make an array [ "yellow-belt", "red-belt", etc] and you can do this:

beltNumber: 0,
belt: function(){ belt_array[belt_number]},
levelUp: function(){ this.belt_number += 1}